osu.Game/Users/ExtendedUserPanel.cs

```diff
@@ -91,37 +92,53 @@ protected FillFlowContainer CreateStatusMessage(bool rightAlignedChildren)
             return statusContainer;
         }
 
-        private void displayStatus(UserStatus? status, UserActivity activity = null)
+        private void updatePresence()
         {
-            if (status != null)
-            {
-                LastVisitMessage.FadeTo(status == UserStatus.Offline && User.LastVisit.HasValue ? 1 : 0);
+            UserPresence? presence;
 
-                // Set status message based on activity (if we have one) and status is not offline
-                if (activity != null && status != UserStatus.Offline)
-                {
-                    statusMessage.Text = activity.GetStatus();
-                    statusMessage.TooltipText = activity.GetDetails();
-                    statusIcon.FadeColour(activity.GetAppropriateColour(Colours), 500, Easing.OutQuint);
-                    return;
-                }
+            if (User.Equals(api?.LocalUser.Value))
+                presence = new UserPresence { Status = api.Status.Value, Activity = api.Activity.Value };
+            else
+                presence = metadata?.GetPresence(User.OnlineID);
```

Comment:
Would you be against this following the standard server flow rather than a special case for the local user?

ie. Make it so the metadata server will always send the local user's presence updates back to them so they know when it's applied on the server side? personally i think this will *feel* better (i'd expect my own user card to update based on when the status is ack'd, rather than when applied).